src/olympia/addons/management/commands/process_addons.py

```diff
@@ -145,7 +146,19 @@ def get_recalc_needed_filters():
         'qs': [Q(migrated_from_lwt__isnull=False,
                  type=amo.ADDON_STATICTHEME, status=amo.STATUS_APPROVED,
                  addonapprovalscounter__last_content_review__isnull=True)]
-    }
+    },
+    'delete_obsolete_addons': {
+        'method': delete_addons,
+        'qs': [
+            Q(type__in=(amo.ADDON_THEME,
+                        amo.ADDON_LPADDON,
+                        amo.ADDON_PLUGIN,
+                        9,  # ADDON_PERSONA,
+                        11,  # ADDON_WEBAPP,
+                        ))
+        ],
+        'allowed_kwargs': ('with_deleted',),
```

Comment:
IMHO the behavior of the task would be clearer if the argument was called `hard_delete` and not `with_deleted`. `with_deleted` makes sense in `process_addons` because it's used to fetch stuff, but it makes less sense in the task where we're using `unfiltered` anyway. So instead of passing it with `allowed_kwargs`, rename it and pass it through `kwargs` ?

Reply:
I wrote it that way originally.  But I couldn't see a use-case where you wouldn't also need `with_deleted` when using `hard_delete`, because you otherwise couldn't hard-delete anything previously soft-deleted otherwise, and for anything not soft-deleted you'd want to soft-delete first to attempt file system clean-up, etc.  
So I removed `hard_delete` and just relied on `with_deleted` - I can add it back though.

Reply:
I see. It's fine, it's just a little confusing because the name doesn't really suit the task. I guess just adding a comment in the task will be enough.